ext/socket/lib/socket.rb

```diff
@@ -593,6 +594,229 @@ def accept_nonblock(exception: true)
     __accept_nonblock(exception)
   end
 
+  GETADDRINFO_V6_DONE = 1
+  GETADDRINFO_V4_DONE = 2
+  RESOLUTION_DELAY_DONE = 3
+  private_constant :GETADDRINFO_V6_DONE, :GETADDRINFO_V4_DONE, :RESOLUTION_DELAY_DONE
+
+  def self.start_getaddrinfo_v6(host, port, mutex, pipe_write, ai_list_v6, error_queue)
+    return Thread.new do
+      Thread.current.report_on_exception = false
+      ai_list_v6.replace(Addrinfo.getaddrinfo(host, port, :PF_INET6, :STREAM))
```

Comment:
Instead of Addrinfo, should these changes be using IPSocket.getaddress? This way, a custom DNS resolver using Resolv and [resolv-replace](https://github.com/ruby/ruby/blob/d92f09a5eea009fa28cd046e9d0eb698e3d94c5c/lib/resolv-replace.rb) should work?